`yarn check-types` builds a production bundle and creates untracked files

Since we merged https://github.com/ChainSafe/lodestar/pull/6685, running `yarn check-types` has become really slow and at the same time also creates untracked files (see https://github.com/ChainSafe/lodestar/pull/6685#discussion_r1586631639)

My suggestion is to either revert the PR or come up with a proper solution as of right now this negatively affects developer experience.

It also fails randomly locally, and even in the CI (see https://github.com/ChainSafe/lodestar/pull/6685#discussion_r1586635384)


- [x] fix build error
- [x] no more untracked files as side effect of running build
- [x] `yarn lint` should consistently pass, irrespective of running `yarn check-types` before
- [x] move build out of `yarn check-types` to speed up time to run `yarn check-types` by  25%-30%

Noticed something else, running `yarn lint` before `yarn check-types` causes no issues. But if `check-types` is executed once locally, any lint runs will fail with the following
```
/home/nico/projects/ethereum/lodestar/packages/light-client/test/unit/webEsmBundle.browser.test.ts
  1:1  error  Unused eslint-disable directive (no problems were reported from '@typescript-eslint/no-unsafe-call')
```

--- a/packages/light-client/vite.config.ts
+++ b/packages/light-client/vite.config.ts
@@ -1,5 +1,5 @@
 import path from "node:path";
-import fs from "node:fs/promises";
+import fs from "node:fs";
 import {defineConfig, mergeConfig} from "vite";
 import dts from "vite-plugin-dts";
 import {getBaseViteConfig} from "../../vite.base.config.js";
@@ -13,8 +13,8 @@ export default mergeConfig(
       dts({
         rollupTypes: true,
         bundledPackages: ["@lodestar/*", "@chainsafe/persistent-merkle-tree", "@chainsafe/bls", "@chainsafe/ssz"],
-        async afterBuild() {
-          await fs.rename(
+        afterBuild() {
+          fs.renameSync(
             path.join(import.meta.dirname, "dist", "index.d.ts"),
             path.join(import.meta.dirname, "dist", "lightclient.min.d.mts")
           );

--- a/vite.base.config.ts
+++ b/vite.base.config.ts
@@ -33,7 +33,7 @@ export function getBaseViteConfig(
         globals: {Buffer: true, process: true},
         protocolImports: true,
       }),
-      ...(process.env.CI ? [] : [visualizer()]),
+      ...(process.env.DEBUG_BUNDLE ? [visualizer()] : []),
     ],
     mode: "production",
     appType: "custom",

